browser/ui/brave_file_select_helper_unittest.cc

```diff
@@ -0,0 +1,213 @@
+       "This page wants to open"},
+      // - subframe:
+      {"file subframe", "http://foo.com/", false, "file:///path/to/page.html",
+       "An embedded page on this page wants to open"},
+
+      // data:
+      // /!\ NOTE that this is for data URLs entered directly in the omnibox.
+      // For pages that generate frames with data URLs, see the browsertest.
+      // - main frame:
+      {"data main frame", "data:blahblah", true, "", "This page wants to open"},
+      // - subframe:
+      {"data subframe", "http://foo.com/", false, "data:blahblah",
+       "An embedded page on this page wants to open"},
+
+      // javascript:
+      // /!\ NOTE that this is for javascript URLs entered directly in the
+      // omnibox. For pages that generate frames with javascript URLs, see the
+      // browsertest.
+      // - main frame:
+      {"javascript main frame", "javascript:abc", true, "",
+       "This page wants to open"},
+      // - subframe:
+      {"javascript subframe", "http://foo.com/", false, "javascript:abc",
+       "An embedded page on this page wants to open"},
+
+      // about:
+      // /!\ NOTE that this is for about:blank URLs entered directly in the
+      // omnibox. For pages that generate frames with about:blank URLs, see the
+      // browsertest.
+      // - main frame:
+      {"about main frame", "about:blank", true, "", "This page wants to open"},
+      // - subframe:
+      {"about subframe", "http://foo.com/", false, "about:blank",
+       "An embedded page on this page wants to open"},
+
+      // blob:
+      // - main frame:
+      {"blob main frame",
+       "blob:http://foo.com/66666666-6666-6666-6666-666666666666", true, "",
+       "foo.com wants to open"},
+      // - subframe:
+      {"blob subframe", "http://bar.com/", false,
+       "blob:http://foo.com/66666666-6666-6666-6666-666666666666",
+       "An embedded page on foo.com wants to open"},
+
+      // filesystem:
+      // - main frame:
+      {"filesystem main frame", "filesystem:http://foo.com/bar.html", true, "",
+       "foo.com wants to open"},
+      // - subframe:
+      {"filesystem subframe", "http://bar.com/", false,
+       "filesystem:http://foo.com/bar.html",
+       "An embedded page on foo.com wants to open"},
+  };
+  for (const auto& test_case : kCases) {
+    SCOPED_TRACE(test_case.case_name);
+    url::Origin main_frame_origin =
+        url::Origin::Create(GURL(test_case.main_frame_url));
+    url::Origin alerting_frame_origin =
+        test_case.is_main_frame
+            ? main_frame_origin
+            : url::Origin::Create(GURL(test_case.alerting_frame_url));
+    EXPECT_EQ(
+        base::UTF8ToUTF16(test_case.expected),
+        brave::GetSiteFrameTitleForFileSelect(
+            brave::GetSiteFrameTitleType(main_frame_origin,
+                                         alerting_frame_origin),
+            alerting_frame_origin, brave::GetFileSelectResourceIDsForOpen()));
```

Comment:
is this missing `ForSave` tests?

also the locale should be enforced in the test suite when you're testing localizable strings. see `brave_l10n::test::ScopedDefaultLocale`.

Reply:
Nice! Such a convenient function